sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs

```diff
@@ -220,12 +224,16 @@ public RehydrationToken GetRehydrationToken()
             string headerSource,
             string? lastKnownLocation,
             string finalStateVia,
-            string? operationId = null)
+            string operationId)
         {
-            var data = new BinaryData(new { version = RehydrationTokenVersion, id = operationId, requestMethod = requestMethod.ToString(), initialUri = startRequestUri.AbsoluteUri, nextRequestUri, headerSource, lastKnownLocation, finalStateVia });
+            // TODO: Once we remove NextLinkOperationImplementation from internal shared and make it internal to Azure.Core only, we can access the internal members from RehydrationToken directly
+            var json = $"{{\"version\":\"{RehydrationTokenVersion}\",\"id\":\"{operationId}\",\"requestMethod\":\"{requestMethod}\",\"initialUri\":\"{startRequestUri.AbsoluteUri}\",\"nextRequestUri\":\"{nextRequestUri}\",\"headerSource\":\"{headerSource}\",\"finalStateVia\":\"{finalStateVia}\",\"lastKnownLocation\":{ConstructStringValue(lastKnownLocation)}}}";
```

Comment:
You could make this more readable by using ` """ `:

```suggestion
            var json = $$"""
            {"version":"{{RehydrationTokenVersion}}","id":"{{operationId}}","requestMethod":"{{requestMethod}}","initialUri":"{{startRequestUriAbsoluteUri}}","nextRequestUri":"{{nextRequestUri}}","headerSource":"{{headerSource}}","finalStateVia":"{{finalStateVia}}","lastKnownLocation":{{ConstructStringValue(lastKnownLocation)}}}
            """;
```

Reply:
looks good, updated